tensorboard/webapp/runs/store/runs_reducers.ts

```diff
@@ -208,6 +208,48 @@ const dataReducer: ActionReducer<RunsDataState, Action> = createReducer(
       groupKeyToColorString,
     };
   }),
+  on(
+    runsActions.runGroupByChanged,
+    (state: RunsDataState, {experimentIds, groupBy}): RunsDataState => {
+      // Reset the groupKeyToColorString
+      const groupKeyToColorString = new Map<string, string>();
+      const defaultRunColorForGroupBy = new Map(
+        state.defaultRunColorForGroupBy
+      );
+
+      const allRuns = experimentIds
+        .flatMap((experimentId) => {
+          return state.runIds[experimentId];
+        })
+        .map((runId) => {
+          return state.runMetadata[runId];
+        })
+        .filter(Boolean);
```

Comment:
IIUC, Line 224-226 return a list of Run, which is defined here
https://cs.opensource.google/tensorflow/tensorboard/+/master:tensorboard/webapp/runs/store/runs_types.ts;l=30-36?q=RunsDataState&ss=tensorflow%2Ftensorboard
What does the `filter(Boolean)` do? Can the list include invalid runs? (like None?)

Reply:
Ah thanks a lot for catching this. I was wrong in thinking that the type would be `undefined` because I am accessing a random property on an object but indeed I was wrong; TypeScript does not complain even if I remove `filter(Boolean)`. Removed it accordingly.